src/adtypes.jl

```diff
@@ -0,0 +1,219 @@
+"""
+AutoEnzyme <: AbstractADType
+An AbstractADType choice for use in OptimizationFunction for automatically
+generating the unspecified derivative functions. Usage:
+```julia
+OptimizationFunction(f, AutoEnzyme(); kwargs...)
+```
+This uses the [Enzyme.jl](https://github.com/EnzymeAD/Enzyme.jl) package. Enzyme performs automatic differentiation on the LLVM IR code generated from julia.
+It is highly-efficient and its ability perform AD on optimized code allows Enzyme to meet or exceed the performance of state-of-the-art AD tools.
+  - Compatible with GPUs
+  - Compatible with Hessian-based optimization
+  - Compatible with Hv-based optimization
+  - Compatible with constraints
+Note that only the unspecified derivative functions are defined. For example,
+if a `hess` function is supplied to the `OptimizationFunction`, then the Hessian
+is not defined via Enzyme.
+"""
```

Comment:
This should be in ADTypes?

Reply:
I thought since that might be used in other packages having this here, for the Optimization specific use cases might make sense?

Reply:
If you are okay with this, this PR is ready

Reply:
I don't think we should do this kind of piracy. Defining docstrings for objects that live in a different library feels like a slippery slope. What happens if two libraries do this? We might need a better system.

Reply:
I am wondering if it's any different from having docstrings for methods of a function in different packages? I am assuming that since the docstrings will be shown if Optimization is loaded, so even if another package adds docstrings to the same struct it shouldn’t be an issue?